src/symmetric_crypto/key_wrapping_rfc_5649.rs

```diff
@@ -0,0 +1,498 @@
+use aead::{generic_array::GenericArray, KeyInit};
+use aes_gcm::aes::{Aes128, Aes192, Aes256};
+use chacha20::cipher::{BlockDecrypt, BlockEncrypt};
+
+use crate::CryptoCoreError;
+
+#[allow(dead_code)]
+const DEFAULT_IV: u64 = 0xA6A6_A6A6_A6A6_A6A6;
+#[allow(dead_code)]
+const DEFAULT_RFC5649_CONST: u32 = 0xA659_59A6_u32;
+
+/// Build the iv according to the RFC 5649
+///
+/// `data_size` is the size of the key to wrap
+#[allow(dead_code)]
+fn build_iv(data_size: usize) -> u64 {
+    let l = u64::from(DEFAULT_RFC5649_CONST);
+    let r = u32::to_le(data_size as u32);
+
+    l << 32 | u64::from(r)
```

Comment:
I think you can safely use `as` to convert:
```suggestion
    (DEFAULT_RFC5649_CONST as u64) << 32 | ((data_size as u32).to_le() as u64)
```